recovery/worker/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"go.temporal.io/temporal/client"
+	"go.temporal.io/temporal/worker"
+	"go.temporal.io/temporal/workflow"
+	"go.uber.org/zap"
+
+	"github.com/temporalio/temporal-go-samples/recovery"
+	"github.com/temporalio/temporal-go-samples/recovery/cache"
+)
+
+func main() {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+
+	// The client and worker are heavyweight objects that should be created once per process.
+	c, err := client.NewClient(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		logger.Fatal("Unable to create client", zap.Error(err))
+	}
+
+	ctx := context.WithValue(context.Background(), recovery.TemporalClientKey, c)
+	ctx = context.WithValue(ctx, recovery.WorkflowExecutionCacheKey, cache.NewLRU(10))
+
+	workflowWorker := worker.New(c, "recovery-task-list", worker.Options{
+		Logger:                    logger,
+		BackgroundActivityContext: ctx,
+	})
+
+	workflowWorker.RegisterWorkflowWithOptions(recovery.RecoverWorkflow, workflow.RegisterOptions{Name: "RecoverWorkflow"})
+	workflowWorker.RegisterWorkflowWithOptions(recovery.TripWorkflow, workflow.RegisterOptions{Name: "TripWorkflow"})
+	workflowWorker.RegisterActivity(recovery.ListOpenExecutions)
+
+	err = workflowWorker.Start()
+	if err != nil {
+		logger.Fatal("Unable to start worker", zap.Error(err))
+	}
+
+	hostWorker := worker.New(c, recovery.HostID, worker.Options{
```

Comment:
This seems not working correctly. And I don't understand how worker for random task list (`HostID` is random) can be used.

Reply:
@mfateev can you try this sample after I merge it?

Reply:
I think I fixed it.